spec/unit/commands/service_command/create_command_spec.rb

```diff
@@ -0,0 +1,95 @@
+ require '3scale_toolbox'
+
+RSpec.describe ThreeScaleToolbox::Commands::ServiceCommand::ServiceCreateSubcommand do
+  include_context :random_name
+
+  context '#run' do
+    let(:remote) { double('remote') }
+    let(:options) { {} }
+
+    subject { described_class.new(options, arguments, nil) }
+
+    before :example do
+      expect(subject).to receive(:threescale_client).with('myremote').and_return(remote)
+    end
+
+    # This should cover the case where the service already exists
+    # or other errors that are returned by calls to the API
+    context "when there is an error creating the service" do
+      let(:arguments) { {remote: "myremote", service_name: "existingservice"} }
+      let(:exists_error_response) { { 'errors' => { 'system_name' => ["has already been taken"] } } }
+
+      it 'an error is raised' do
+        expect(remote).to receive(:create_service).and_return(exists_error_response)
+        expect do
+          subject.run
+        end.to raise_error(ThreeScaleToolbox::Error, /Service has not been saved/)
+      end
+    end
+
+    context "service name parameter is specified" do
+      let(:arguments) { {remote: "myremote", service_name: service_name} }
+      let(:service_name) { "a_service_name" }
+      let(:service_create_args) { {"name" => service_name} }
+      let(:service_create_result) { {"name" => service_name, "system_name" => service_name, "id" => "1" } }
+
+      after :example do
```

Comment:
Although still valid, `after` hook should be used to do cleanup kind of tasks.

If you want to share some expectation and DRY, the [shared_examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples) are the way to go.

Reply:
:+1: changed